### PR TITLE
KFSPTS-29084 implement Jaggaer generate XML report

### PR DESCRIPTION
**Pull request by jhulslander**
There is a related non-prod-sql PR for this change
ConcurBatchUtilityService. getFileContents was deprecated.  The function was copied into LoadFileUtils,  This was done because this functionality is useful in multiple areas.  I didn't want to refactor everything that uses ConcurBatchUtilityService.getFileContents to avoid scope creep for this change.

I changed the LOG level in a catch block as it was just adding "cruff" to the job LOG file.  Other logging in the same area provide details needed if debugging is needed.

### CODE REVIEW

File: src/main/java/edu/cornell/kfs/module/purap/batch/service/impl/JaggaerGenerateSupplierXmlReportServiceImpl.java
Comment (cah292) [inline src/main/java/edu/cornell/kfs/module/purap/batch/service/impl/JaggaerGenerateSupplierXmlReportServiceImpl.java:127]: There are several log statements in this method that need to update the mentioned method name: `sendEmail` --> `emailReport`.

Also, if desired, the debug statements could be parameterized and the call to `isDebugEnabled()` could be removed.

File: src/main/java/edu/cornell/kfs/module/purap/batch/service/impl/JaggaerGenerateSupplierXmlReportServiceImpl.java
Comment (cah292) [inline src/main/java/edu/cornell/kfs/module/purap/batch/service/impl/JaggaerGenerateSupplierXmlReportServiceImpl.java:129]: Minor typo in message: `budy` --> `body`

File: src/main/java/edu/cornell/kfs/module/purap/batch/service/impl/JaggaerGenerateSupplierXmlReportServiceImpl.java
Comment (cah292) [inline src/main/java/edu/cornell/kfs/module/purap/batch/service/impl/JaggaerGenerateSupplierXmlReportServiceImpl.java:137]: Is logging this message at info level okay, or should it be logged at warning or error level instead?

File: src/main/java/edu/cornell/kfs/module/purap/batch/service/impl/JaggaerUploadFileServiceImpl.java
Comment (cah292) [inline src/main/java/edu/cornell/kfs/module/purap/batch/service/impl/JaggaerUploadFileServiceImpl.java:202]: Minor typo in method name: `...FileResusts` --> `...FileResults`

Also, is this method supposed to be converted into a report-generating one? Or has that work been deferred to a future ticket?

File: src/main/java/edu/cornell/kfs/sys/util/LoadFileUtils.java
Comment (cah292) [inline src/main/java/edu/cornell/kfs/sys/util/LoadFileUtils.java:57]: I know the original method only used the single-arg String constructor, but you may want to consider updating this to use a constructor that also accepts a charset.

File: src/main/resources/edu/cornell/kfs/module/purap/cu-purap-resources.properties
Comment (cah292) [inline src/main/resources/edu/cornell/kfs/module/purap/cu-purap-resources.properties:10]: I see that the tab-prefixed messages also have a single space immediately after the tabs. Is that intentional, or should such lines be prefixed with only tabs or only spaces?

**Review by cah292 — CHANGES_REQUESTED** [review]
Changes look good overall, though I did notice some areas for further cleanup.

File: src/main/java/edu/cornell/kfs/concur/batch/service/ConcurBatchUtilityService.java
Comment (jhulslander) [reply src/main/java/edu/cornell/kfs/concur/batch/service/ConcurBatchUtilityService.java:142]: ok

File: src/main/java/edu/cornell/kfs/module/purap/batch/service/impl/JaggaerGenerateSupplierXmlReportServiceImpl.java
Comment (jhulslander) [reply src/main/java/edu/cornell/kfs/module/purap/batch/service/impl/JaggaerGenerateSupplierXmlReportServiceImpl.java:78]: ok

File: src/main/java/edu/cornell/kfs/module/purap/batch/service/impl/JaggaerGenerateSupplierXmlReportServiceImpl.java
Comment (jhulslander) [reply src/main/java/edu/cornell/kfs/module/purap/batch/service/impl/JaggaerGenerateSupplierXmlReportServiceImpl.java:86]: ok

File: src/main/java/edu/cornell/kfs/module/purap/batch/service/impl/JaggaerGenerateSupplierXmlReportServiceImpl.java
Comment (jhulslander) [reply src/main/java/edu/cornell/kfs/module/purap/batch/service/impl/JaggaerGenerateSupplierXmlReportServiceImpl.java:127]: ok

File: src/main/java/edu/cornell/kfs/module/purap/batch/service/impl/JaggaerGenerateSupplierXmlReportServiceImpl.java
Comment (jhulslander) [reply src/main/java/edu/cornell/kfs/module/purap/batch/service/impl/JaggaerGenerateSupplierXmlReportServiceImpl.java:129]: ok

File: src/main/java/edu/cornell/kfs/module/purap/batch/service/impl/JaggaerGenerateSupplierXmlReportServiceImpl.java
Comment (jhulslander) [reply src/main/java/edu/cornell/kfs/module/purap/batch/service/impl/JaggaerGenerateSupplierXmlReportServiceImpl.java:137]: good point, I will switch to warn

File: src/main/java/edu/cornell/kfs/module/purap/batch/service/impl/JaggaerUploadFileServiceImpl.java
Comment (jhulslander) [reply src/main/java/edu/cornell/kfs/module/purap/batch/service/impl/JaggaerUploadFileServiceImpl.java:202]: I switched the name to logUploadFileResults

The functionals currently believe they don't need this particular report.  Jaggaer generates a report internally if there are errors on the upload.  We should continue to log the results to the job run log to aid in trouble shooting if there are upload errors.

File: src/main/java/edu/cornell/kfs/sys/util/LoadFileUtils.java
Comment (jhulslander) [reply src/main/java/edu/cornell/kfs/sys/util/LoadFileUtils.java:57]: ok

**Comment by jhulslander** [comment]
this is ready for review again

File: src/main/java/edu/cornell/kfs/module/purap/batch/service/impl/JaggaerUploadFileServiceImpl.java
Comment (cah292) [reply src/main/java/edu/cornell/kfs/module/purap/batch/service/impl/JaggaerUploadFileServiceImpl.java]: Ok, thanks for the explanation.

File: src/main/resources/edu/cornell/kfs/module/purap/cu-purap-resources.properties
Comment (cah292) [reply src/main/resources/edu/cornell/kfs/module/purap/cu-purap-resources.properties]: Thanks for the clarification. I'm fine with the tabs and spaces being left in as-is.

File: src/main/java/edu/cornell/kfs/sys/util/LoadFileUtils.java
Comment (cah292) [inline src/main/java/edu/cornell/kfs/sys/util/LoadFileUtils.java:53]: Did you mean to say "UTF_8" in this comment?

**Review by cah292 — CHANGES_REQUESTED** [review]
I noticed a few more items that may need cleanup, but otherwise the changes look good.

File: src/main/java/edu/cornell/kfs/module/purap/batch/service/impl/JaggaerGenerateSupplierXmlReportServiceImpl.java
Comment (jhulslander) [reply src/main/java/edu/cornell/kfs/module/purap/batch/service/impl/JaggaerGenerateSupplierXmlReportServiceImpl.java:53]: ok

File: src/main/java/edu/cornell/kfs/module/purap/batch/service/impl/JaggaerGenerateSupplierXmlReportServiceImpl.java
Comment (jhulslander) [reply src/main/java/edu/cornell/kfs/module/purap/batch/service/impl/JaggaerGenerateSupplierXmlReportServiceImpl.java:78]: ok

File: src/main/java/edu/cornell/kfs/module/purap/batch/service/impl/JaggaerGenerateSupplierXmlReportServiceImpl.java
Comment (jhulslander) [reply src/main/java/edu/cornell/kfs/module/purap/batch/service/impl/JaggaerGenerateSupplierXmlReportServiceImpl.java:122]: oh, thank you, I forgot about that, I am going to switch the from address

File: src/main/java/edu/cornell/kfs/module/purap/batch/service/impl/JaggaerGenerateSupplierXmlReportServiceImpl.java
Comment (jhulslander) [reply src/main/java/edu/cornell/kfs/module/purap/batch/service/impl/JaggaerGenerateSupplierXmlReportServiceImpl.java:141]: ok

File: src/main/java/edu/cornell/kfs/sys/util/LoadFileUtils.java
Comment (jhulslander) [reply src/main/java/edu/cornell/kfs/sys/util/LoadFileUtils.java:53]: yes, will fix

**Comment by jhulslander** [comment]
this is ready for review again

**Review by cah292 — COMMENTED** [review]
Latest changes look good. I see that the unit tests supposedly passed, but due to technical issues with the instance that runs the tests, I'm wondering if it would be best to wait for that to be resolved before I merge this, just to confirm that the tests did indeed run.

**Comment by cah292** [comment]
As per the recent team meeting, we're confident that the tests executed and passed as expected. I will proceed with the merge.

**Pull request merged**
